TodoList/TodoList.Api/Controllers/NodesController.cs

```diff
@@ -21,16 +21,16 @@ public NodesController()
         public async Task<IHttpActionResult> GetAsync()
             => await Task.FromResult<IHttpActionResult>(Ok(Nodes));
 
-        public async Task<IHttpActionResult> GetAsync(string id)
+        public async Task<IHttpActionResult> GetAsync(Guid id)
             => await Task.FromResult<IHttpActionResult>(Ok(Nodes[0]));
 
         public async Task<IHttpActionResult> PostAsync([FromBody] string text) 
             => await Task.FromResult<IHttpActionResult>(CreatedAtRoute("Nodes", new { id = Nodes[1].Id.ToString() }, Nodes[1]));
 
-        public async Task<IHttpActionResult> PutAsync(string id, string text)
+        public async Task<IHttpActionResult> PutAsync(Guid id, string text)
```

Comment:
The text is sensitive information, therefore you should use the [FromBody] annotation. The HTTPs protocol is able to cipher the body of the request but not the URL address.
Anyway, you should send the whole NodeModel in the PutAsync header (instead of its attributes) since it is more scalable. (Also, you should do the same in the PostAsync method + check whether the sent id is default(Guid))